vispy/visuals/polygon.py

```diff
@@ -38,17 +36,34 @@ class PolygonVisual(Visual):
     """
     def __init__(self, pos=None, color='black',
                  border_color=None, border_width=1, **kwargs):
-        super(PolygonVisual, self).__init__(**kwargs)
-
-        self.mesh = MeshVisual()
-        self.border = LineVisual()
+        self._mesh = MeshVisual()
+        self._border = LineVisual()
         self._pos = pos
         self._color = Color(color)
         self._border_width = border_width
         self._border_color = Color(border_color)
         self._update()
-        #glopts = kwargs.pop('gl_options', 'translucent')
-        #self.set_gl_options(glopts)
+
+        CompoundVisual.__init__(self, [self._mesh, self._border], **kwargs)
+
+    def _update(self):
```

Comment:
This code only needs to be called at most once per draw. I would add something like:
```
def _prepare_draw(self, view):
    if self._need_update:
        self._update()  
        # or you could even combine the two methods 
        # rather than calling _update here
```
..and change all the existing calls to `_update()` to look like:
```
    self._need_update = True
    self.update()
```